test/functional/rpc_net.py

```diff
@@ -139,8 +145,21 @@ def _test_getaddednodeinfo(self):
         # check that a non-existent node returns an error
         assert_raises_rpc_error(-24, "Node has not been added", self.nodes[0].getaddednodeinfo, '1.1.1.1')
 
-    def _test_getpeerinfo(self):
+    def test_getpeerinfo(self):
+        self.log.info("Test getpeerinfo")
+        if self.is_wallet_compiled():
+            self.nodes[0].sendtoaddress(self.nodes[1].getnewaddress(), 1)
+            self.nodes[1].generate(1)
+            self.sync_all()
+        time_now = int(time.time())
         peer_info = [x.getpeerinfo() for x in self.nodes]
+        # Verify last_block and last_transaction values.
+        for x in range(self.num_nodes):
```

Comment:
Consider using [itertools.product()](https://docs.python.org/2/library/itertools.html#itertools.product) rather than a triple-nested for loop here. It'd also be helpful if there were more meaningful variable names than `x` and `y`.

Reply:
Good idea -- done. It's much nicer now.